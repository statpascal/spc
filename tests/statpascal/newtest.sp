program dummy;

const n = 10;

var a: ^integer;
    s: ^string;
    t: string;
    i: 0..n;

begin
    new (a, n);
    new (s, n);
    for i := 0 to n - 1 do begin
        a [i] := i + 1;
        str (a [i], t);
        s [i] := 'Hello, world #' + t
    end;
    for i := 0 to n - 1 do
        writeln (a [i], ': ', s [i]);

    dispose (a);
    dispose (s);
end.
