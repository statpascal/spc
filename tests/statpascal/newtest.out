1: Hello, world #1
2: Hello, world #2
3: Hello, world #3
4: Hello, world #4
5: Hello, world #5
6: Hello, world #6
7: Hello, world #7
8: Hello, world #8
9: Hello, world #9
10: Hello, world #10
